### PR TITLE
Auth implementation

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**

### CODE REVIEW

File: src/services/api/lib.ts [hunk 1]
```diff
@@ -1,44 +1,72 @@
...
   });
 };
 
+export const signUpNewUser = async ({
+  username,
+  permissions,
+  email,
+}: CreateUserProp): Promise<AuthToken & UserData> => {
+  return axios
+    .post(endpoints.USER_SIGN_UP(), {
```
Comment (Extheoisah) [inline src/services/api/lib.ts:33]: You need to pass the github token in the headers as `x-github-token` for new users. That's what the backend uses to create the jwt that is returned

File: src/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -44,60 +41,55 @@ export const authOptions: NextAuthOptions = {
...
-        email,
-        permissions,
-      }: CreateUserProp) => {
-        const res = await createNewUser({ username, email, permissions });
-        if (res.data) {
+
+      if (isNewUser && profile?.login) {
+        const { email, login } = profile;
+        try {
+          const signedUpUser = await signUpNewUser({
```
Comment (Extheoisah) [inline src/pages/api/auth/[...nextauth].ts:48]: this function should receive the GitHub token

File: src/services/api/lib.ts (hunk 1 above)
Comment (Emmanuel-Develops) [reply src/services/api/lib.ts:33]: good catch

File: src/pages/api/auth/[...nextauth].ts (hunk 2 above)
Comment (Emmanuel-Develops) [reply src/pages/api/auth/[...nextauth].ts:48]: yup yup, I'll address that

**Review by Extheoisah — APPROVED** [review]

**Pull request merged**